### ISSUE TITLE
got error with fla for function using stl

### ISSUE THREAD

**Issue by funkvps** [za233/Polaris-Obfuscator#7]
After several test, I found a crash bug, when using fla for functions using stl.

Below is an example

int __attribute__((annotate("fla"))) initHook()
{
	if (MH_Initialize() == MH_OK)
	{
		char name[0x104];
		GetModuleFileNameA(NULL, name, 0x104);
		std::string exedllName = PathFindFileNameA(name);
		PrintDebugLog("MH_Initialize Succ");
		return 0;
	}
	else
	{
		PrintDebugLog("MH_Initialize failed");
		return 1;
	}
}

Below is error

1>This block has 1 successor
1>This block has 1 successor
1>This block has 1 successor
1>This block has 1 successor
1>fatal error: error in backend: Cannot select: 0x1240e6060f8: i32 = X86ISD::CMP BasicBlock:ch< 0x1240e9a0600>, Constant:ch<0>
1>In function: ?initHook@@YAHXZ
1>PLEASE submit a bug report to https://github.com/llvm/llvm-project/issues/ and include the crash backtrace, preprocessed source, and associated run script.
1>Stack dump:
1>0.	Program arguments: "D:\\Program Files\\Microsoft Visual Studio\\2022\\Community\\VC\\Tools\\Llvm\\x64\\bin\\clang-cl.exe" -cc1 -triple amd64-pc-windows-msvc19.41.34123 -emit-obj -mincremental-linker-compatible -disable-free -clear-ast-before-backend -disable-llvm-verifier -discard-value-names -main-file-name dllmain.cpp -mrelocation-model pic -pic-level 2 -mframe-pointer=none -relaxed-aliasing -fmath-errno -ffp-contract=on -fno-rounding-math -mconstructor-aliases -funwind-tables=2 -target-cpu x86-64 -mllvm -x86-asm-syntax=intel -tune-cpu generic -D_MT -D_DLL --dependent-lib=msvcrt --dependent-lib=oldnames -stack-protector 2 -fcxx-exceptions -fexceptions -fexternc-nounwind -fms-volatile -fdefault-calling-conv=cdecl -fdiagnostics-format msvc -gno-column-info -gcodeview -debug-info-kind=constructor -ffunction-sections -fcoverage-compilation-dir=C:\\Users\\cools\\source\\repos\\ProMaxLoaderNew\\netapi32 -resource-dir "D:\\Program Files\\Microsoft Visual Studio\\2022\\Community\\VC\\Tools\\Llvm\\x64\\lib\\clang\\17" -D NDEBUG -D NETAPI32_EXPORTS -D _WINDOWS -D _USRDLL -D _WINDLL -D _UNICODE -D UNICODE -internal-isystem "D:\\Program Files\\Microsoft Visual Studio\\2022\\Community\\VC\\Tools\\Llvm\\x64\\lib\\clang\\17\\include" -internal-isystem "D:\\Program Files\\Microsoft Visual Studio\\2022\\Community\\VC\\Tools\\Llvm\\x64\\lib\\clang\\17\\include" -internal-isystem "D:\\Program Files\\Microsoft Visual Studio\\2022\\Community\\VC\\Tools\\MSVC\\14.41.34120\\include" -internal-isystem "D:\\Program Files\\Microsoft Visual Studio\\2022\\Community\\VC\\Tools\\MSVC\\14.41.34120\\atlmfc\\include" -internal-isystem "D:\\Program Files\\Microsoft Visual Studio\\2022\\Community\\VC\\Auxiliary\\VS\\include" -internal-isystem "D:\\Windows Kits\\10\\Include\\10.0.22621.0\\ucrt" -internal-isystem "D:\\Windows Kits\\10\\Include\\10.0.22621.0\\um" -internal-isystem "D:\\Windows Kits\\10\\Include\\10.0.22621.0\\shared" -internal-isystem "D:\\Windows Kits\\10\\Include\\10.0.22621.0\\winrt" -internal-isystem "D:\\Windows Kits\\10\\Include\\10.0.22621.0\\cppwinrt" -internal-isystem "C:\\Program Files (x86)\\Windows Kits\\NETFXSDK\\4.8\\Include\\um" -internal-isystem "D:\\Program Files\\Microsoft Visual Studio\\2022\\Community\\VC\\Tools\\MSVC\\14.41.34120\\include" -internal-isystem "D:\\Program Files\\Microsoft Visual Studio\\2022\\Community\\VC\\Tools\\MSVC\\14.41.34120\\atlmfc\\include" -internal-isystem "D:\\Program Files\\Microsoft Visual Studio\\2022\\Community\\VC\\Auxiliary\\VS\\include" -internal-isystem "D:\\Windows Kits\\10\\Include\\10.0.22621.0\\ucrt" -internal-isystem "D:\\Windows Kits\\10\\Include\\10.0.22621.0\\um" -internal-isystem "D:\\Windows Kits\\10\\Include\\10.0.22621.0\\shared" -internal-isystem "D:\\Windows Kits\\10\\Include\\10.0.22621.0\\winrt" -internal-isystem "D:\\Windows Kits\\10\\Include\\10.0.22621.0\\cppwinrt" -internal-isystem "C:\\Program Files (x86)\\Windows Kits\\NETFXSDK\\4.8\\Include\\um" -O2 -Wall -Wno-error -fdeprecated-macro -fdebug-compilation-dir=C:\\Users\\cools\\source\\repos\\ProMaxLoaderNew\\netapi32 -ferror-limit 19 -fno-use-cxa-atexit -fms-extensions -fms-compatibility -fms-compatibility-version=19.41.34123 -std=c++14 -fdelayed-template-parsing -fno-caret-diagnostics -vectorize-loops -vectorize-slp -faddrsig -object-file-name=C:\\Users\\cools\\source\\repos\\ProMaxLoaderNew\\netapi32\\x64\\Release\\dllmain.obj -o x64\\Release\\dllmain.obj -x c++ dllmain.cpp
1>1.	<eof> parser at end of file
1>[obf] run.registerOptimizerLastEPCallback
1>2.	Code generation
1>3.	Running pass 'Function Pass Manager' on module 'dllmain.cpp'.
1>4.	Running pass 'X86 DAG->DAG Instruction Selection' on function '@"?initHook@@YAHXZ"'

**Comment by za233** [comment]
This obfuscator pass is from OLLVM, which is an old and classic obfuscation framework. So there may be some bugs. Improved flattening pass will be added later.

**Comment by funkvps** [comment]
> This obfuscator pass is from OLLVM, which is an old and classic obfuscation framework. So there may be some bugs. Improved flattening pass will be added later.

Yes, I just tested several differnet repo, fla all failed when compile.  

such as https://github.com/DreamSoule/ollvm17

**Comment by funkvps** [comment]
Problem root cause found. If I disable exception support for visual studio project, then fla works OK. it seems current fla implementation does not compatible with c++ exception.
